### PR TITLE
DEVPROD-6041 Documentation generation improvements.

### PR DESCRIPTION
**Pull request by ahartschen**
Thanks for submitting a PR to the Genny repo. Please include the following fields (if relevant) prior to submitting your PR.

**Jira Ticket:** DEVPROD-6041

**Whats Changed:**  
1. Made separate documentation files for workloads and phases.
2. Added hyperlinks to the respective workload and phase yamls in the docs.

### CODE REVIEW

File: src/lamplib/src/genny/templates/documentation.md.j2 [hunk 1]
```diff
@@ -1,6 +1,6 @@
-# Workload Documenation
+# Workload Documentation 
 {% for workload in workloads %}
```
Comment (dalyd) [inline src/lamplib/src/genny/templates/documentation.md.j2:2]: Can we add a line break inside this loop, either at the top or bottom? I think 
```
---
```
Would be sufficient.

**Review by dalyd — APPROVED** [review]
The change LGTM. I have two requests that can be in this PR or separate. 
1. Can we add a horizontal line between entries? 
2. Can we link to the generated docs from https://github.com/mongodb/genny/blob/3765880c20407c696af4c56568fa932b4940c342/docs/start.md?

File: src/lamplib/src/genny/templates/documentation.md.j2 [hunk 2]
```diff
@@ -1,6 +1,6 @@
-# Workload Documenation
+# Workload Documentation
```
Comment (SimonEismann) [inline src/lamplib/src/genny/templates/documentation.md.j2:1]: The added space here makes me think that we aren't linting this. That's probably okay in the commit queue, but running some linter over this locally might be a good idea.

File: docs/generated/workloads.md [hunk 3]
```diff
@@ -1,925 +1,6 @@
-# Workload Documenation
+# Workload Documentation
```
Comment (SimonEismann) [inline docs/generated/workloads.md:1]: We need some introductory text here:
* What workloads does this cover?
* What information is available for each workload?
* Where does this documentation come from?
* What to do if documentation is missing/bad?
* ...

File: docs/generated/workloads.md
Comment (SimonEismann) [inline docs/generated/workloads.md:1]: This is a bit ambitious, so feel free to ignore.

We could have two sections here:
* Active Workloads (workloads with an autorun section that run on sys-perf)
* Other Workloads (the rest)

My concern is that users might find a workload on this page and then go looking for it in sys-perf and not find it.

**Review by SimonEismann — CHANGES_REQUESTED** [review]

File: src/lamplib/src/genny/templates/documentation.md.j2 (hunk 1 above)
Comment (dalyd) [reply src/lamplib/src/genny/templates/documentation.md.j2:2]: Ah, yeah -- the heading including the line does make that confusing.

**Review by SimonEismann — APPROVED** [review]

File: docs/generated/workloads.md
Comment (SimonEismann) [reply docs/generated/workloads.md:1]: Done: https://jira.mongodb.org/browse/DEVPROD-6160

**Pull request merged**